src/geneinfo.js

```diff
@@ -86,6 +102,88 @@ router.get('/api/gene/:gene', async (ctx) => {
   });
 });
 
+
+router.get('/api/genes/', async (ctx) => {  
+
+  var genesString = ctx.query.genes;
+  var genes = genesString.split(",")
+
+  var source = ctx.query.source;
+  var species = ctx.query.species;
+  var build = ctx.query.build;
+  if (source == null || source == '') {
+    source = 'gencode';
+  } 
+  var geneSqlString = "SELECT distinct * from genes where gene_name ";
+  geneSqlString    += getGenesInClause(genes);
+  geneSqlString    += " AND source = \""+source+"\"";
+  if (species != null && species != "") {
+    geneSqlString  += " AND species = \""+species+"\"";
+  }
+  if (build != null && build != "") {
+    geneSqlString  += " AND build = \""+build+"\"";
+  }
+
+  return new Promise((resolve, reject) => {
+    db.all(geneSqlString,function(err,rows){ 
+      var gene_data = {};
+
+      if (rows != null && rows.length > 0) {
+        for (var i = 0; i < rows.length; i++) {
+          let gene_record = rows[i]
+          gene_record.transcripts = []
+          gene_data[gene_record.gene_name] = gene_record;    
+        }
+      } 
+          
+      var source = ctx.query.source; 
+      if (source == null || source == '') {
+        source = 'gencode';
+      } 
+      var sqlString = "";
+      if (source == 'gencode') {
+        sqlString =  "SELECT t.*, x.refseq_id as 'xref' from transcripts t ";
+        sqlString += "LEFT OUTER JOIN xref_transcript x on x.gencode_id = t.transcript_id ";
+      } else if (source == 'refseq') {
+        sqlString =  "SELECT t.*, x.gencode_id as 'xref' from transcripts t ";
+        sqlString += "LEFT OUTER JOIN xref_transcript x on x.refseq_id = t.transcript_id ";
+      }
+      sqlString     += "WHERE t.gene_name ";
+      sqlString     += getGenesInClause(genes);
+      sqlString +=    "AND t.source = \""+source+"\"";
+      if (species != null && species != "") {
+        sqlString  += " AND t.species = \""+species+"\"";
+      }
+      if (build != null && build != "") {
+        sqlString  += " AND t.build = \""+build+"\"";
+      }        
+      db.all(sqlString,function(err,transcriptRows){    
+
+        if (err) reject(err);
+
+
+        if (transcriptRows != null && transcriptRows.length > 0) {
+          for (var i = 0; i < transcriptRows.length; i++) {
+            var transcript = transcriptRows[i]; 
+            transcript['features'] = JSON.parse(transcript['features']);
+            gene_record = gene_data[transcript.gene_name]
+            if (gene_record) {
+              gene_record["transcripts"].push(transcript);
+            } else {
+              console.log("cannot find gene for transcript " + transcript.transcript_id + " " + transcript.gene_name)
+            }  
+          }
+        }
+
+        ctx.set('Content-Type', 'application/json');
+        ctx.set('Charset', 'utf-8')
+        ctx.body = ctx.query.callback + '(' + JSON.stringify([gene_data]) +');';
```

Comment:
Since this is just a GET request with standard headers, we shouldn't need to use JSONP here. I think we can just do `ctx.body = JSON.stringify([gene_data])`

Reply:
Okay.  Let me try that.